inc/Engine/CriticalPath/CriticalCSSGeneration.php

```diff
@@ -54,52 +54,36 @@ public function __construct( ProcessorService $processor ) {
 	 * @return bool false if task performed successfully, true otherwise to re-queue the item.
 	 */
 	protected function task( $item ) {
```

Comment:
We can't type cast unfortunately, because the `task()` is extending from the parent class.

`Declaration of WP_Rocket\Engine\CriticalPath\CriticalCSSGeneration::task(array $item) must be compatible with WP_Background_Process::task($item)`

Reply:
@Tabrisrp `task` has to be compatible with the background handler's implementation. In that cause, we should check if `$item` is an array. If no, what should we do?

Reply:
Receiving something else than an array is unexpected, IMO we should bail out early and return false